rust/lance/src/dataset/write/merge_insert.rs

```diff
@@ -515,17 +522,30 @@ impl MergeInsertJob {
 
         // Apply deletions
         let removed_row_ids = Arc::into_inner(deleted_rows).unwrap().into_inner().unwrap();
+
         let (old_fragments, removed_fragment_ids) =
             Self::apply_deletions(&self.dataset, &removed_row_ids).await?;
 
         // Commit updated and new fragments
-        Self::commit(
+        let committed_ds = Self::commit(
             self.dataset,
             removed_fragment_ids,
             old_fragments,
             new_fragments,
         )
-        .await
+        .await?;
+
+        let stats = Arc::into_inner(merge_statistics)
+            .unwrap()
+            .into_inner()
+            .unwrap();
+
+        Ok((
+            committed_ds,
+            stats.num_inserted_rows,
+            stats.num_updated_rows,
+            stats.num_deleted_rows,
```

Comment:
Similarly here, why return these as separate values instead of as the `MergeStats` struct?